src/nvim/lua/vim.lua

```diff
@@ -478,4 +478,19 @@ function vim.defer_fn(fn, timeout)
   return timer
 end
 
+--- Get effective tab size.
+---
+--- This is the number of visual columns the cursor moves over when the <Tab>
+--- key is pressed in insert mode.
+---
+--@see |softtabstop|
+--@param bufnr (optional, number): Buffer handle, defaults to current
+--@return (number) tabsize
+function vim.get_effective_tabsize(bufnr)
```

Comment:
Does that mean that the information provided by this function to the LSP server can be wrong in some cases? Or are the ways in which it can go wrong irrelevant for the LSP use case?

If the former, then the implementation should probably be refined; if the latter, then reducing its advertised scope by moving it out of the `vim` namespace sounds like a good idea.

Reply:
I think that the logic is fine for the `vim` namespace, but maybe the name and docs should be a bit different. I initially suggested `get_effective_tabsize`, but it actually only deals with tabstops and not tabsizes. So maybe if we named it `get_effective_tabstop`, it would be fine, because I think it does cover the entire scope of tabstops. I'm of course also fine with just putting it in `lsp.util` first.

@h-michael

Reply:
Let's move to lsp.util and rename to get_effective_tabstops.

Reply:
Sounds good to me :) I'll make the change, if there aren't any objections.

Reply:
D'oh, how did I miss that. Thanks!

(Selective blindness, I think I was expecting `test/functional/runtime`. And I found some `runtime/lua/vim/uri.lua` tests under `test/functional/lua`, so I didn't think to look elsewhere.)

Reply:
Done in e1d7ae7c5857038718167bfc1da6a0e809fdf4cb, please let me know if you spot any issues :)